use latest ubuntu 18.04 for gitlab runner instances

roock opened the pull request:
## Description
The default image used for the docker-machine EC2 instances is rather old. In this PR this is changed to use the latest AMI of Ubuntu 18.04 available on AWS EC2.

## Migrations required
NO

## Verification
Works without problems in our installation for a few weeks

npalm commented:
@roock thanks for the improvement. I will try it the next days. Aksed my self the question should a be a configuration parameter or not?

npalm commented:
@roock works like a charm, please can you have a look on the one comment I made. Just about the naming.

roock commented:
@npalm which comment about naming do you mean?
about adding a parameter for this: This came also to my mind, I'll try to add it to the MR

npalm commented:
@roock I mean the name of the terraform resource 

```data "aws_ami" "latest-ubuntu" {
 ``

I think name the data resource `docker-machine-ami` is a more clear name. And indeed having the filter configurable would be great.

roock commented:
@npalm  I've added a parameter for the docker machine ami defaulting to latest Ubuntu 16.04. Or should we set default to Ubuntu 18.04?
also changed the name of the terraform resource as requested.

npalm commented on variables.tf:
```diff
@@ -315,6 +315,22 @@ variable "ami_owners" {
+variable "runner_ami_owners" {
+  description = "The list of owners used to select the AMI of Gitlab runner docker-machine instances."
+  type        = "list"
+  default     = ["Canonical"]
```
A look up by name is not supported, I get the following error
```
Error: Error refreshing state: 1 error occurred:
	* module.runner.data.aws_ami.docker-machine: 1 error occurred:
	* module.runner.data.aws_ami.docker-machine: data.aws_ami.docker-machine: InvalidUserID.Malformed: Invalid user id: "Canonical"
	status code: 400, request id: 
```

Replacing Canonical by 099720109477 does the trick. Suggest that you replace Canonical by the id and add a comment line that it is the id of Canonical

roock replied:
fixed in b08e1f8e5cb641c8d6ceb95cff2ce292bce1aee7

roock commented on main.tf:
```diff
@@ -198,6 +199,22 @@ resource "aws_autoscaling_group" "gitlab_runner_instance" {
   ]
 }
 
+data "aws_ami" "latest-ubuntu" {
```
already changed to "docker-machine"

npalm commented:
@roock Cool for creating this PR. I dit a quick check and now the runners are using `ubuntu/images/hvm-ssd/ubuntu-xenial-16.04-amd64-server-20180228.1 (ami-0b541372)` instead of `ubuntu/images/hvm-ssd/ubuntu-xenial-16.04-amd64-server-20190628 (ami-03746875d916becc0)
`. Which means it looks like the default behaviour is not matching the docs at the docker site (https://docs.docker.com/machine/drivers/aws/) Which mention the default should be the daily ubuntu image.
